src/components/policy/policy_external/test/sql_pt_representation_storage_test.cc

```diff
@@ -0,0 +1,121 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "policy/sql_pt_representation.h"
+#include "utils/file_system.h"
+#include "policy/sql_wrapper.h"
+#include "policy/mock_policy_settings.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+
+#include <string>
+#include <memory>
+
+using testing::ReturnRef;
+using testing::NiceMock;
+
+namespace test {
+namespace components {
+namespace policy_test {
+
+class SQLPTRepresentationStorageTest : public ::testing::Test {
+ protected:
+  const std::string kAppStorageFolder =
+      "storage_SQLPTRepresentationStorageTest";
+
+  std::shared_ptr<policy::SQLPTRepresentation> reps;
+  std::shared_ptr<utils::dbms::SQLQuery> query_wrapper_;
+  std::shared_ptr<policy_handler_test::MockPolicySettings> policy_settings_;
+
+  void SetUp() {
+    file_system::CreateDirectory(kAppStorageFolder);
+
+    reps = std::make_shared<policy::SQLPTRepresentation>();
+    ASSERT_TRUE(reps != NULL);
+
+    policy_settings_ =
+        std::make_shared<NiceMock<policy_handler_test::MockPolicySettings> >();
+    ON_CALL(*policy_settings_, app_storage_folder())
+        .WillByDefault(ReturnRef(kAppStorageFolder));
+    EXPECT_EQ(::policy::SUCCESS, reps->Init(policy_settings_.get()));
+
+    query_wrapper_ = std::make_shared<utils::dbms::SQLQuery>(reps->db());
+    ASSERT_TRUE(query_wrapper_ != NULL);
+  }
+
+  void TearDown() OVERRIDE {
+    EXPECT_TRUE(reps->Drop());
+    EXPECT_TRUE(reps->Close());
+    reps->RemoveDB();
+
+    policy_settings_.reset();
+
+    file_system::remove_directory_content(kAppStorageFolder);
+    file_system::RemoveDirectory(kAppStorageFolder, true);
+  }
+};
+
+TEST_F(
+    SQLPTRepresentationStorageTest,
+    CheckPermissionsAllowed_SetValuesInAppGroupRpcFunctionalGroup_GetEqualParamsInCheckPermissionResult) {
+  // Arrange
+  const std::string query =
+      "INSERT OR REPLACE INTO `application` (`id`, `memory_kb`,"
+      " `heart_beat_timeout_ms`) VALUES ('12345', 5, 10); "
+      "INSERT OR REPLACE INTO functional_group (`id`, `name`)"
+      "  VALUES (1, 'Base-4'); "
+      "INSERT OR REPLACE INTO `app_group` (`application_id`,"
+      " `functional_group_id`) VALUES ('12345', 1); "
+      "INSERT OR REPLACE INTO `rpc` (`name`, `parameter`, `hmi_level_value`,"
+      " `functional_group_id`) VALUES ('Update', 'gps', 'FULL', 1); "
+      "INSERT OR REPLACE INTO `rpc` (`name`, `parameter`, `hmi_level_value`,"
+      " `functional_group_id`) VALUES ('Update', 'speed', 'FULL', 1);";
+
+  // Assert
+  ASSERT_TRUE(query_wrapper_->Exec(query));
+
+  // Act
+  policy::CheckPermissionResult ret;
+  reps->CheckPermissions("12345", "FULL", "Update", ret);
```

Comment:
@ZhdanovP suggest to move all these constants to a separate variables and use them in `query` text and compare their values here

Reply:
@AKalinich-Luxoft I used the approach that are using in other tests of this project